CMakeLists.txt

```diff
@@ -0,0 +1,163 @@
+project(os-autoinst CXX)
+cmake_minimum_required(VERSION 3.3.0)
+
+# list non-C++ source files
+set(DOC_FILES
+    doc/architecture.md
+    doc/backend_vars.asciidoc
+    doc/backends.md
+    doc/memorydumps.asciidoc
+    doc/networking.md
+)
+set(PERL_FILES
+    autotest.pm
+    backend/amt.pm
+    backend/baseclass.pm
+    backend/console_proxy.pm
+    backend/driver.pm
+    backend/generalhw.pm
+    backend/ikvm.pm
+    backend/ipmi.pm
+    backend/null.pm
+    backend/pvm_hmc.pm
+    backend/pvm.pm
+    backend/qemu.pm
+    backend/s390x.pm
+    backend/spvm.pm
+    backend/svirt.pm
+    backend/virt.pm
+    basetest.pm
+    bmwqemu.pm
+    commands.pm
+    consoles/amtSol.pm
+    consoles/console.pm
+    consoles/ipmiSol.pm
+    consoles/localXvnc.pm
+    consoles/network_console.pm
+    consoles/remoteVnc.pm
+    consoles/s3270.pm
+    consoles/serial_screen.pm
+    consoles/sshIucvconn.pm
+    consoles/ssh_screen.pm
+    consoles/sshVirtsh.pm
+    consoles/sshVirtshSUT.pm
+    consoles/sshX3270.pm
+    consoles/sshXtermIPMI.pm
+    consoles/sshXtermVt.pm
+    consoles/ttyConsole.pm
+    consoles/virtio_terminal.pm
+    consoles/vnc_base.pm
+    consoles/VNC.pm
+    distribution.pm
+    lockapi.pm
+    mmapi.pm
+    myjsonrpc.pm
+    needle.pm
+    ocr.pm
+    OpenQA/Benchmark/Stopwatch.pm
+    OpenQA/Commands.pm
+    OpenQA/Exceptions.pm
+    OpenQA/Isotovideo/CommandHandler.pm
+    OpenQA/Isotovideo/Interface.pm
+    OpenQA/Isotovideo/NeedleDownloader.pm
+    OpenQA/Isotovideo/Utils.pm
+    OpenQA/Qemu/BlockDevConf.pm
+    OpenQA/Qemu/BlockDev.pm
+    OpenQA/Qemu/ControllerConf.pm
+    OpenQA/Qemu/DriveController.pm
+    OpenQA/Qemu/DriveDevice.pm
+    OpenQA/Qemu/DrivePath.pm
+    OpenQA/Qemu/MutParams.pm
+    OpenQA/Qemu/PFlashDevice.pm
+    OpenQA/Qemu/Proc.pm
+    OpenQA/Qemu/SnapshotConf.pm
+    OpenQA/Qemu/Snapshot.pm
+    OpenQA/Test/RunArgs.pm
+    osutils.pm
+    signalblocker.pm
+    testapi.pm
+)
+set(MISC_FILES
+    consoles/icewm.cfg
+    crop.py
+    dmidata/dell_e6330/smbios_type_1.bin
+    dmidata/dell_e6330/smbios_type_2.bin
+    dmidata/dell_e6330/smbios_type_3.bin
+    dmidata/dump
+)
+
+# determine install directories
+# note: Not using ${CMAKE_INSTALL_LIBDIR} here because this seems always lib in our packaging (and
+#       never lib64) regardless of the architecture.
```

Comment:
Shouldn't it be `CMAKE_INSTALL_FULL_LIBDIR` even?

And why is it called "data" dir? If these are supposed to be architecture agnostic it should probably be `CMAKE_INSTALL_FULL_DATADIR`. :thinking:

Reply:
> Shouldn't it be CMAKE_INSTALL_FULL_LIBDIR even?

No, it shouldn't. This is used with the `DESTINATION` option of `install()` and the documentation states explicitly not to use the full version here: https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html#result-variables

> And why is it called "data" dir?

This directory contains various files so I didn't want to be more specific (Perl modules, a executable, a Python script, some "dell_e6330" binaries). I tried to be consistent and compatible with the current build system where this directory is e.g. `/usr/lib/os-autoinst`. Hence it makes no sense to use `CMAKE_INSTALL_DATADIR`.